src/main/java/io/candydoc/ddd/extract_ddd_concepts/DDDConceptsExtractionService.java

```diff
@@ -64,72 +70,95 @@ private void trackAndApply(List<Event> occurredEvents) {
   }
 
-  public void handle(ExtractDomainCommands command) {
-    log.info("Extract domain commands from {}", command.getPackageToScan());
-    trackAndApply(domainCommandExtractor.extract(command));
+  public void handle(ExtractSharedKernels command) {}
+
+  public void handle(ExtractValueObjects command) {
+    log.info("Extract value objects from {}", command.getPackageToScan());
+    trackAndApply(valueObjectExtractor.extract(command));
   }
 
   public void handle(CheckConceptInteractions command) {
     log.info("Check concept interactions from {}", command.getConceptName());
     trackAndApply(interactionChecker.check(command));
   }
 
-  public void handle(ExtractAggregates command) {
-    log.info("Extract aggregates from {}", command.getPackageToScan());
-    trackAndApply(aggregatesExtractor.extract(command));
+  public void apply(AggregateFound event) {
+    this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
   public void apply(BoundedContextFound event) {
+    this.handle(ExtractAggregates.builder().packageToScan(event.getPackageName()).build());
     this.handle(ExtractCoreConcepts.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractDomainEvents.builder().packageToScan(event.getPackageName()).build());
     this.handle(ExtractDomainCommands.builder().packageToScan(event.getPackageName()).build());
-    this.handle(ExtractAggregates.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractDomainEvents.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
+    this.handle(
+        CheckConceptInteractions.builder()
+            .conceptName(event.getPackageName() + ".package-info")
+            .build());
   }
 
   public void apply(CoreConceptFound event) {
     this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
-  public void apply(InteractionBetweenConceptFound event) {}
-
-  public void apply(ValueObjectFound event) {
+  public void apply(DomainCommandFound event) {
     this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
   public void apply(DomainEventFound event) {
     this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
   }
 
-  public void apply(DomainCommandFound event) {
-    this.handle(CheckConceptInteractions.builder().conceptName(event.getCanonicalName()).build());
+  public void apply(SharedKernelFound event) {
+    this.handle(ExtractCoreConcepts.builder().packageToScan(event.getPackageName()).build());
+    this.handle(ExtractValueObjects.builder().packageToScan(event.getPackageName()).build());
+    this.handle(
+        CheckConceptInteractions.builder()
```

Comment:
Same as the bounded context, put the Canonical name in the event

Reply:
Fixed in commit https://github.com/Candy-Doc/candy-doc-maven-plugin/pull/129/commits/b60080ee6c99bf5e558bfaf38d8ddc78e42bf3be